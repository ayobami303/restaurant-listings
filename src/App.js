import React from 'react';
import Button from './component/common/Button';
import Input from './component/common/Input';
import Checkbox from './component/common/Checkbox';
import Dropdown from './component/common/Dropdown';
import Label from './component/common/Label';

function App() {
    return(
        <div className="App container">
            Hello world
            <Button/>
            <Label />
            <Input/>
            <Checkbox />
            <Dropdown />
        </div>
	);
}

export default App;
