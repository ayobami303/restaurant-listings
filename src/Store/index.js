import React, { createContext, useState, useContext, useEffect } from 'react';
import { api, getCategories, getRestaurantListing } from "../api";


export const GlobalStoreContext = createContext({
  locations: [],
  options: {},
  sortType: "Location",
  value: "",
  message: "Enter Location to search",
  restaurantResult: [],
  categories: []
})

export const useMyContext = () => useContext(GlobalStoreContext);

function ContextProvider ({children}) {
  const [ globalStore, setGlobalStore ] = useState({
    locations: [],
    options: {},
    sortType: "Location",
    value: "",
    message: "Enter Location to search",
    restaurantResult: [],
    categories: []
  });

  useEffect(() => {
    const fetchCategories = async () => {
      try {
        const [ categories, restaurants ] = await Promise.all([
          getCategories(),
          getRestaurantListing(),
        ]);

        setGlobalStore(prev => ({ ...prev, restaurantResult: restaurants.restaurants, categories: categories.categories }));
      } catch (error) {
        console.log(error)
      }
    }
    fetchCategories();
  }, [])

  return (
    <GlobalStoreContext.Provider value={[globalStore, setGlobalStore]}>
      {children}
    </GlobalStoreContext.Provider>
  );
}
 
export default ContextProvider;
