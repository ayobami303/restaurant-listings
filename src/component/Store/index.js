import React, { createContext, useState } from 'react';

export const GlobalStoreContext = createContext({
  locations: [],
  options: {},
  sortType: "Location",
  value: "",
  message: "Enter Location to search",
  restaurantResult: [],
})

function Store ({children}) {
  const [ globalStore, setGlobalStore ] = useState({
    locations: [],
    options: {},
    sortType: "Location",
    value: "",
    message: "Enter Location to search",
    restaurantResult: [],
  });

  return (
    <GlobalStoreContext.Provider value={[globalStore, setGlobalStore]}>
      {children}
    </GlobalStoreContext.Provider>
   );
}
 
export default Store;
