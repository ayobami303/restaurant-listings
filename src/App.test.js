import React from "react";
import ReactDOM from "react-dom";
import { App } from "./App";
import Store from "./Store";

const Index = () => {
  return (
    <Store>
      <App />
    </Store>
  );
};

export default Index;
describe('', () => {
  beforeAll(() => {
    jest.spyOn(console, 'error').mockImplementation((...args) => {
      if (!args[0].includes('Warning: An update to %s inside a test was not wrapped in act')) {
        console.error(...args);
      }
    });
  });
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Index />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
