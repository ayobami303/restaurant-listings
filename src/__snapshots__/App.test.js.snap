// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
"<div className=\\"container\\">
  <div className=\\"row m-5\\">
    <div className=\\"col\\">
      <BrowserRouter>
        <Switch>
          <Route path=\\"/\\">
            <div className=\\"row\\">
              <div className=\\"col hello\\">
                <_class />
              </div>
            </div>
          </Route>
        </Switch>
      </BrowserRouter>
    </div>
  </div>
</div>"
`;
